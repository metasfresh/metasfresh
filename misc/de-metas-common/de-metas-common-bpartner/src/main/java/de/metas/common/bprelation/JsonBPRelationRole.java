/*
 * #%L
 * de.metas.business.rest-api
 * %%
 * Copyright (C) 2021 metas GmbH
 * %%
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as
 * published by the Free Software Foundation, either version 2 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public
 * License along with this program. If not, see
 * <http://www.gnu.org/licenses/gpl-2.0.html>.
 * #L%
 */

package de.metas.common.bprelation;

import de.pentabyte.springfox.ApiEnum;

public enum JsonBPRelationRole
{
	@ApiEnum("Hauptlieferant")
	MainProducer,
	@ApiEnum("Krankenhaus / Hostpital")
	Hospital,
	@ApiEnum("Arzt / Doctor")
	PhysicianDoctor,
	@ApiEnum("Hausarzt / General practitioner")
	GeneralPractitioner,
	@ApiEnum("Krankenkasse / Health Insurance")
	HealthInsurance,
	@ApiEnum("Pflegeheim / Nursing home")
	NursingHome,
	@ApiEnum("Betreuer / Caregiver")
	Caregiver,
	
	@ApiEnum("Patientenkontakt / Caretaker")
	CareTaker,
	
	@ApiEnum("Wunschapotheke / Preferred Pharmacy")
	PreferredPharmacy,
	@ApiEnum("Pflegedienst / Nursing service")
	NursingService,
	@ApiEnum("Payer / Payer")
	Payer,
	@ApiEnum("Apotheke / Pharmacy")
	Pharmacy
}
