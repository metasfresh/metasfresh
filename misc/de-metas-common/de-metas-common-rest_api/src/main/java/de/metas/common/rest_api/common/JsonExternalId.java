/*
 * #%L
 * de-metas-common-rest_api
 * %%
 * Copyright (C) 2021 metas GmbH
 * %%
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as
 * published by the Free Software Foundation, either version 2 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public
 * License along with this program. If not, see
 * <http://www.gnu.org/licenses/gpl-2.0.html>.
 * #L%
 */

package de.metas.common.rest_api.common;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import de.metas.common.util.EmptyUtil;
import lombok.NonNull;
import lombok.Value;

import javax.annotation.Nullable;
import java.util.Objects;

@Value
public class JsonExternalId
{
	String value;

	@JsonCreator
	public static JsonExternalId of(@NonNull final String value)
	{
		return new JsonExternalId(value);
	}

	@Nullable
	public static JsonExternalId ofOrNull(@Nullable final String value)
	{
		if (EmptyUtil.isBlank(value))
		{
			return null;
		}
		return new JsonExternalId(value);
	}

	private JsonExternalId(@NonNull final String value)
	{
		if (value.trim().isEmpty())
		{
			throw new RuntimeException("Param value=" + value + " may not be empty");
		}
		this.value = value;
	}

	@JsonValue
	public String getValue()
	{
		return value;
	}

	public static boolean equals(@Nullable final JsonExternalId id1, @Nullable final JsonExternalId id2)
	{
		return Objects.equals(id1, id2);
	}

	@Nullable
	public static String toValue(@Nullable final JsonExternalId externalId)
	{
		if (externalId == null)
		{
			return null;
		}
		return externalId.getValue();
	}

}
