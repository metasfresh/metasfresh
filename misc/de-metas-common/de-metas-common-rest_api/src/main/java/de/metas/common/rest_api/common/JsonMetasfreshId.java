/*
 * #%L
 * de-metas-common-rest_api
 * %%
 * Copyright (C) 2021 metas GmbH
 * %%
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as
 * published by the Free Software Foundation, either version 2 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public
 * License along with this program. If not, see
 * <http://www.gnu.org/licenses/gpl-2.0.html>.
 * #L%
 */

package de.metas.common.rest_api.common;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import lombok.NonNull;
import lombok.Value;

import javax.annotation.Nullable;
import java.util.Objects;

@Value
public class JsonMetasfreshId
{
	int value;

	@JsonCreator
	public static JsonMetasfreshId of(@NonNull final Integer value)
	{
		return new JsonMetasfreshId(value);
	}

	@Nullable
	public static JsonMetasfreshId ofOrNull(@Nullable final Integer value)
	{
		if (isEmpty(value))
		{
			return null;
		}
		return new JsonMetasfreshId(value);
	}

	private static boolean isEmpty(@Nullable final Integer value)
	{
		return value == null || value < 0;
	}

	private JsonMetasfreshId(@NonNull final Integer value)
	{
		if (value < 0)
		{
			throw new RuntimeException("Given value=" + value + " needs to be >=0");
		}
		this.value = value;
	}

	@JsonValue
	public int getValue()
	{
		return value;
	}

	public static boolean equals(@Nullable final JsonMetasfreshId id1, @Nullable final JsonMetasfreshId id2)
	{
		return Objects.equals(id1, id2);
	}

	@Nullable
	public static Integer toValue(@Nullable final JsonMetasfreshId externalId)
	{
		if (externalId == null)
		{
			return null;
		}
		return externalId.getValue();
	}

	public static int toValueInt(@Nullable final JsonMetasfreshId externalId)
	{
		if (externalId == null)
		{
			return -1;
		}
		return externalId.getValue();
	}

	public static String toValueStr(@Nullable final JsonMetasfreshId externalId)
	{
		if (externalId == null)
		{
			return "-1";
		}
		return String.valueOf(externalId.getValue());
	}
}
