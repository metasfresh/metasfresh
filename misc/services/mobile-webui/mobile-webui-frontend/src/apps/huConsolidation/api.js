import axios from 'axios';
import { toUrl, unboxAxiosResponse } from '../../utils';
import { apiBasePath } from '../../constants';

export const getAvailableTargets = ({ wfProcessId }) => {
  return axios
    .get(toUrl(`${apiBasePath}/mobile/huConsolidation/job/${wfProcessId}/target/available`))
    .then((response) => unboxAxiosResponse(response));
};

export const setTarget = ({ wfProcessId, target }) => {
  return axios
    .post(toUrl(`${apiBasePath}/mobile/huConsolidation/job/${wfProcessId}/target`), target)
    .then((response) => unboxAxiosResponse(response));
};

export const closeTarget = ({ wfProcessId }) => {
  return axios
    .post(toUrl(`${apiBasePath}/mobile/huConsolidation/job/${wfProcessId}/target/close`))
    .then((response) => unboxAxiosResponse(response));
};

export const consolidate = ({ wfProcessId, fromPickingSlotQRCode, huId }) => {
  return axios
    .post(toUrl(`${apiBasePath}/mobile/huConsolidation/job/${wfProcessId}/consolidate`), {
      wfProcessId,
      fromPickingSlotQRCode,
      huId,
    })
    .then((response) => unboxAxiosResponse(response));
};

export const getPickingSlotContent = ({ wfProcessId, pickingSlotId }) => {
  return axios
    .get(`${apiBasePath}/mobile/huConsolidation/job/${wfProcessId}/pickingSlot/${pickingSlotId}`)
    .then((response) => unboxAxiosResponse(response));
};
