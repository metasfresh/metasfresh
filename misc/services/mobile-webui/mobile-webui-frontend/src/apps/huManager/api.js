import axios from 'axios';
import { apiBasePath } from '../../constants';
import { unboxAxiosResponse } from '../../utils';
import { toQRCodeString } from '../../utils/huQRCodes';

const huAPIBasePath = `${apiBasePath}/material/handlingunits`;

export const getHUByQRCode = (qrCode) => {
  return axios
    .post(`${huAPIBasePath}/byQRCode`, { qrCode })
    .then(unboxAxiosResponse)
    .then((response) => response.result);
};

export const disposeHU = ({ huId, reasonCode }) => {
  return axios.post(`${huAPIBasePath}/byId/${huId}/dispose?reasonCode=${reasonCode}`).then(unboxAxiosResponse);
};

export const getDisposalReasonsArray = () => {
  return axios
    .get(`${huAPIBasePath}/disposalReasons`)
    .then(unboxAxiosResponse)
    .then((response) => response.reasons);
};

/**
 * @returns {Promise<T>} handling unit info
 */
export const moveHU = ({ huId, huQRCode, targetQRCode }) => {
  return axios
    .post(`${huAPIBasePath}/move`, {
      huId,
      huQRCode: toQRCodeString(huQRCode),
      targetQRCode: toQRCodeString(targetQRCode),
    })
    .then(unboxAxiosResponse)
    .then((response) => response.result);
};

/**
 * @returns {Promise<T>} handling unit info
 */
export function getAllowedClearanceStatusesRequest({ huId }) {
  return axios
    .get(`${huAPIBasePath}/byId/${huId}/allowedClearanceStatuses`)
    .then(unboxAxiosResponse)
    .then((response) => response.clearanceStatuses);
}

export function setClearanceStatusRequest({ huId, clearanceNote = null, clearanceStatus }) {
  return axios.put(`${huAPIBasePath}/byId/${huId}/clearance`, { clearanceStatus, clearanceNote });
}
