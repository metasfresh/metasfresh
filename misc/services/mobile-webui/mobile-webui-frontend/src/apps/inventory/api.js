import axios from 'axios';
import { toUrl, unboxAxiosResponse } from '../../utils';
import { apiBasePath } from '../../constants';

const inventoryBasePath = `${apiBasePath}/mobile/inventory`;

export const resolveLocator = ({ scannedBarcode, wfProcessId, lineId }) => {
  return axios
    .post(toUrl(`${inventoryBasePath}/resolveLocator`), { scannedCode: scannedBarcode, wfProcessId, lineId })
    .then((response) => unboxAxiosResponse(response));
};

export const resolveHU = ({ scannedBarcode, wfProcessId, lineId, locatorQRCode }) => {
  return axios
    .post(toUrl(`${inventoryBasePath}/resolveHU`), { scannedCode: scannedBarcode, wfProcessId, lineId, locatorQRCode })
    .then((response) => unboxAxiosResponse(response));
};
