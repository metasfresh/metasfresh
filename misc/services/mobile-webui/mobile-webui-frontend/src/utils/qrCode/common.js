import { trl } from '../translations';

export const QRCODE_SEPARATOR = '#';

export const parseQRCodeType = (qrCodeString) => {
  const idx = qrCodeString.indexOf(QRCODE_SEPARATOR);
  if (idx <= 0) {
    console.log('onResolvedResult: Cannot extract type from QRCode', { qrCodeString });
    throw trl('error.qrCode.invalid');
  }
  return qrCodeString.substring(0, idx);
};
