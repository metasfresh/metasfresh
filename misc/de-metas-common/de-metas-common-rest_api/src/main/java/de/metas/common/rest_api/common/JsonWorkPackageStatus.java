/*
 * #%L
 * de-metas-common-rest_api
 * %%
 * Copyright (C) 2021 metas GmbH
 * %%
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as
 * published by the Free Software Foundation, either version 2 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public
 * License along with this program. If not, see
 * <http://www.gnu.org/licenses/gpl-2.0.html>.
 * #L%
 */

package de.metas.common.rest_api.common;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import de.metas.common.rest_api.common.JsonMetasfreshId;
import lombok.Builder;
import lombok.NonNull;
import lombok.Value;

import javax.annotation.Nullable;
import java.time.Instant;

@Value
@Builder
public class JsonWorkPackageStatus
{
	JsonMetasfreshId metasfreshId;

	@Nullable
	@JsonInclude(JsonInclude.Include.NON_EMPTY)
	String error;

	Instant enqueued;

	boolean readyForProcessing;

	@Builder
	@JsonCreator
	public JsonWorkPackageStatus(@JsonProperty("metasfreshId") @NonNull final JsonMetasfreshId metasfreshId,
			@JsonProperty("error") @Nullable final String error,
			@JsonProperty("enqueued") final Instant enqueued,
			@JsonProperty("readyForProcessing") final boolean readyForProcessing)
	{
		this.metasfreshId = metasfreshId;
		this.error = error;
		this.enqueued = enqueued;
		this.readyForProcessing = readyForProcessing;
	}
}