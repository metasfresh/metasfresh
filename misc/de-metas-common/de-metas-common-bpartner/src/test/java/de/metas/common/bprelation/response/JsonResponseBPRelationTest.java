/*
 * #%L
 * de.metas.business.rest-api
 * %%
 * Copyright (C) 2021 metas GmbH
 * %%
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as
 * published by the Free Software Foundation, either version 2 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public
 * License along with this program. If not, see
 * <http://www.gnu.org/licenses/gpl-2.0.html>.
 * #L%
 */

package de.metas.common.bprelation.response;

import com.fasterxml.jackson.databind.ObjectMapper;
import de.metas.common.rest_api.common.JsonExternalId;
import de.metas.common.bprelation.JsonBPRelationRole;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

import java.io.IOException;

import static io.github.jsonSnapshot.SnapshotMatcher.expect;
import static io.github.jsonSnapshot.SnapshotMatcher.start;
import static org.assertj.core.api.Assertions.assertThat;

class JsonResponseBPRelationTest
{
	private final ObjectMapper mapper = new ObjectMapper();

	@BeforeAll
	public static void beforeAll()
	{
		start();
	}

	@Test
	public void serializeDeserializeTest() throws IOException
	{
		final JsonResponseBPRelationComposite item = JsonResponseBPRelationComposite.builder()
				.responseItem(JsonResponseBPRelationItem.builder()
						.bpartnerId(123456)
						.locationId(234567)
						.targetBPartnerId(345678)
						.targetBPLocationId(456789)
						.name("relName")
						.description("relDesc")
						.externalId(JsonExternalId.of("ext123"))
						.role(JsonBPRelationRole.MainProducer)
						.shipTo(true)
						.payFrom(true)
						.billTo(true)
						.handoverLocation(true)
						.fetchedFrom(true)
						.remitTo(true)
						.active(true)
						.build())
				.build();

		final String str = mapper.writeValueAsString(item);

		final JsonResponseBPRelationComposite result = mapper.readValue(str, JsonResponseBPRelationComposite.class);
		assertThat(result).isEqualTo(item);

		expect(result).toMatchSnapshot();
	}

}
