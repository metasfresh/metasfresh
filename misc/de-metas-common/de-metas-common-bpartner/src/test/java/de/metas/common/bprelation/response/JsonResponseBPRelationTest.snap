de.metas.common.bprelation.response.JsonResponseBPRelationTest.serializeDeserializeTest=[
  {
    "responseItems": [
      {
        "active": true,
        "billTo": true,
        "bpartnerId": 123456,
        "description": "relDesc",
        "externalId": "ext123",
        "fetchedFrom": true,
        "handoverLocation": true,
        "locationId": 234567,
        "name": "relName",
        "payFrom": true,
        "remitTo": true,
        "role": "MainProducer",
        "shipTo": true,
        "targetBPLocationId": 456789,
        "targetBPartnerId": 345678
      }
    ]
  }
]